streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractPartitionGroup.java

```diff
@@ -56,6 +57,8 @@ abstract class AbstractPartitionGroup {
 
     abstract Long headRecordOffset(final TopicPartition partition);
 
+    abstract Optional<Integer> headRecordLeaderEpoch(final TopicPartition partition);
```

Comment:
Would this really be an `Optional`? In the end, we _always_ need to have some leader epoch -- or do I misunderstand something?

Reply:
leader epoch type is always `Optional<Integer>` (at least across all Consumer code)